src/main/kotlin/io/github/rybalkinsd/kohttp/interceptors/RetryInterceptor.kt

```diff
@@ -0,0 +1,71 @@
+package io.github.rybalkinsd.kohttp.interceptors
+
+import okhttp3.Interceptor
+import okhttp3.Response
+import java.net.SocketTimeoutException
+
+/**
+ * Retry Interceptor
+ *
+ * Retry request if you get an exception or an error code.
+ *
+ * @param failureThreshold number of attempts to get response with default value 3
+ * @param invocationTimeout timeout (millisecond) before retry
+ * @param ratio ratio for exponential increase of invocation timeout
+ * @param errorStatuses error codes that you need to handle with default values 503,504
+ *
+ * @since 0.9.0
+ * @author UDarya
+ * */
+class RetryInterceptor(
+    private val failureThreshold: Int = 3,
+    private val invocationTimeout: Long = 0,
+    private val ratio: Int = 1,
+    errorStatuses: List<Int> = emptyList()
+) : Interceptor {
+    private var _errorStatuses: List<Int> = listOf(503, 504)
```

Comment:
May be it is better to move this into constructor and remove unnecessary field?
And also make it `val`
```suggestion
class RetryInterceptor(
    private val failureThreshold: Int = 3,
    private val invocationTimeout: Long = 0,
    private val ratio: Int = 1,
    private val errorStatuses: List<Int> = listOf(503, 504)
) : Interceptor {
```

Reply:
Here we want to precess 503 and 504 by default and also add some more depending on user provided error statuses.

if `errorStatuses = listOf(426, 500)` -> retry (426, 500, 503, 504)

Suggestions are very welcome